Changes filed_in_one_year values from True/False to Yes/No/Unknown

jenna-anderson opened the pull request:
## Description

**note: filed_in_one_year was changed to check_for_one_year in a previous PR that hadn't been merged yet at the time of this PR creation. The latest version of main was pulled down and merge conflicts were handled and now filed_in_one_year referenced below is actually check_for_one_year, but the logic is the same.

Stakeholder wanted filed_in_one_year values to be yes/no/unknown instead of True/False

- Adds conditional to uploadRouter to change True/False received from DS to yes/no/unknown to be passed to frontend
- Changes filed_in_one_year column type to string instead of boolean to handle conversion from True/False to yes/no/unknown
- Changes seeds for filed_in_one_year from True/False to yes/no/unknown.
-  Update reflected in the ReadMe documentation

** See corresponding PR on frontend for reference: [Frontend PR](https://github.com/Lambda-School-Labs/human-rights-first-asylum-fe-a/pull/353) **

[Trello Card](https://trello.com/c/j0KRCnE3)

Contributors: @schandrupatla 

Fixes issue of the user seeing True/False for Filed Within One Year on the frontend 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

[Loom Video](https://www.loom.com/share/ec0bd68c85714dda9d473a12a5b6104a?sharedAppSource=personal_library)

willmoon19 reviewed (approved):
seems like a simple clean change, good work!

ZacharyCooremans reviewed (approved):
Good changes. Seems like change was easy and well planned out.

jenna-anderson commented:
You're right, @ZacharyCooremans . It looks like it doesn't like the commented out ds_case migration file (the migration file to go with the ds_case seed file that we removed because it was breaking running our seeds). I will try deleting the migration file as well since we've deleted the corresponding seed file since Alex Tran shared with us that these files were not needed but rather for his reference.

This same failed check is occurring in the [alignment of schema PR](https://github.com/Lambda-School-Labs/human-rights-first-asylum-be-a/pull/169). I'll make note of it in that PR and delete the migration file there too, unless you think this could create some sort of merge conflict down the line.
![image](https://user-images.githubusercontent.com/81700660/132705680-329c2cb1-ffff-4081-bbe9-6bf9a7148872.png)
![image](https://user-images.githubusercontent.com/81700660/132706286-2f410db5-1c03-4c9f-8ac2-310c10034bc5.png)

ryan-hamblin reviewed (commented):
Only thing that I saw here that would help for the future development of this project is to add an 'unknown' condition to at least one of the seeds. This will ensure anyone picking up the project has the ability to jump in. 

It was noted that there was live QA done and the 'unknown' flag was tested together in a group!

ryan-hamblin reviewed (approved):
See my comment about the seeds.

ryan-hamblin commented on README.md:
```diff
@@ -53,7 +53,7 @@ The base technologies are JavaScript, HTML and CSS. The frontend leverages [Reac
         "type_of_violence": "Not Applicable",
         "appellate": false,
-        "filed_in_one_year": false,
+        "filed_in_one_year": 'no',
```
Let's standardize on using single OR double quotes. 

Your linter should be catching this, AND your formatter could `autoFormatOnSave` to format to a specific convention.

jenna-anderson replied:
Good catch! We've changed this to double quotes to match the convention of the README file.

joshuascan reviewed (approved):
Looks good to me! I don't see anything that hasn't been mentioned before.

paulstgermain reviewed (approved):
## Video Walkthrough Feedback

Though you explain the task you took and that it came from the Stakeholder, I would have loved a more fully fleshed out explanation regarding why this change was requested by our Stakeholder to lend extra context to your description. 

Your description of the means by which you changed the logic in the `uploadRouter` to translate `true` or `false` values would have been elevated by mentioning that you used a ternary operator verbally - flex that technical knowledge!

I would also have loved to see more detail about the reasons the `ds_cases` seed file needed to be deleted in order to run our seed files. I think that's my main piece of feedback for you here - try to reach beyond the surface when describing these technical decisions moving forward! :) 

## Code Review

The code itself is fairly self-explanatory - you've changed the language used by the application to something more user friendly by changing our use of `true` or `false` to 'yes' or 'no' at our stakeholder's request, and by changing the way we receive this data from the DS API by translating those `true/false` statements before allowing them to be entered into our database you've ensured that choice stays consistent moving forward! You also did well taking feedback and making the appropriate changes, and extensive work was done after this video during a call to work through some merge conflicts and ensure ALL appropriate variables were changed. Well done all around!
